### PR TITLE
Remove iOS 9 support 👋

### PR DESCRIPTION
**Pull request by p4checo**
## Motivation

After some years supporting it, and with iOS 13 right around the corner, we felt it was time to drop support for iOS 9.

It also allows us to remove some legacy code (and tests), and finally be consistent in the iOS target version between main target and test targets.

## Changes

- Set deployment target to iOS 10 at project level, and inherit on all targets.

- Update `s.ios.deployment_target` on .podspec.

- Update Compatibility section in README.md.

- Remove all iOS 9 specific code.

- Revise code to use most up to date primitives.

### CODE REVIEW

File: Sources/Utils/Atomic.swift [hunk 1]
```diff
@@ -32,6 +32,12 @@ public final class Atomic<Value> {
     public var value: Value {
         get { return withValue { $0 } }
         set(newValue) { swap(newValue) }
+        _modify {
```
Comment (renatorodrigues) [inline Sources/Utils/Atomic.swift]: Was this addition intended?

Comment (p4checo) [reply Sources/Utils/Atomic.swift]: Well, it was, even though it's not related to deprecating iOS 9. Technically it only requires Swift 5.0, but I forgot to put it in on the migration PR. 😇 

Do you prefer that I add it on a separate PR?

Comment (renatorodrigues) [reply Sources/Utils/Atomic.swift]: I'd suggest moving this to a separate PR. It's a _little big_ change to be introduced here.

Comment (p4checo) [reply Sources/Utils/Atomic.swift]: Fair enough, will do it 👍
